src/main/org/deidentifier/arx/criteria/EDDifferentialPrivacy.java

```diff
@@ -160,62 +160,59 @@ public int getRequirements(){
         return ARXConfiguration.REQUIREMENT_COUNTER |
                ARXConfiguration.REQUIREMENT_SECONDARY_COUNTER;
     }
-
-    /**
-     * Creates a random sample based on beta
-     *
-     * @param manager
-     */
+    
+    @Override
     public void initialize(DataManager manager, ARXConfiguration config){
         
-        // Needed for consistent de-serialization. We need to call this
-        // method in the constructor of the class DataManager. The following
-        // condition should hold, when this constructor is called during 
-        // de-serialization, when we must not change the subset.
-        if (subset != null && this.manager == null) {
-            this.manager = manager;
+        if (config != null) {
+            // This method was already called within the DataManager class.
             return;
         }
         
-        // Needed to prevent inconsistencies. We need to call this
-        // method in the constructor of the class DataManager. It will be called again, when
-        // ARXConfiguration is initialized(). During the second call we must not change the subset.
-        if (subset != null && this.manager == manager) {
-            return;
+        // Set beta and k if required
+        if (beta < 0) {
+            ParameterCalculation pCalc = null;
+            try {
+                pCalc = new ParameterCalculation(epsilon, delta);
+            } catch (IntervalArithmeticException e) {
+                throw new RuntimeException(e);
+            }
+            beta = pCalc.getBeta();
+            k = pCalc.getK();
         }
         
-        // Calculate beta and k
-        try {
-            ParameterCalculation pCalc = new ParameterCalculation(this.epsilon, this.delta);
-            this.beta = pCalc.getBeta();
-            this.k = pCalc.getK();
-        } catch (IntervalArithmeticException e) {
-            throw new RuntimeException(e);
-        }
-
-        // Create RNG
-        Random random;
-        if (deterministic) {
-            random = new Random(0xDEADBEEF);
-        } else {
-            random = new SecureRandom();
-        }
+        // Perform random sampling iff the subset is null
+        // or if this method has already been called.
+        // Otherwise, we know that this instance was just de-serialized
+        // and that no subsequent call of anonymize() has occured.
+        if (subset == null || initialized) {

```

Comment:
Perform random sampling iff the model is used for the first time (subset == null) or when it used again (initialized == true). We don't perform random sampling when the model has been deserialized (subset will be !=null and initialized will be false).